packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -2,97 +2,40 @@
  * External dependencies
  */
 import classnames from 'classnames';
-import { last } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { useSelect } from '@wordpress/data';
-import { useState, useRef, useEffect, useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	useState,
+	useEffect,
+	useCallback,
+	useRef,
+	useMemo,
+} from '@wordpress/element';
 import { Popover } from '@wordpress/components';
-import { placeCaretAtVerticalEdge } from '@wordpress/dom';
+import { isRTL } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import Inserter from '../inserter';
-import { getClosestTabbable } from '../writing-flow';
 import { getBlockDOMNode } from '../../utils/dom';
 
-function InsertionPointInserter( {
-	clientId,
-	setIsInserterForced,
-	containerRef,
-} ) {
-	const ref = useRef();
-	// Hide the inserter above the selected block and during multi-selection.
-	const isInserterHidden = useSelect(
-		( select ) => {
-			const {
-				getMultiSelectedBlockClientIds,
-				getSelectedBlockClientId,
-				hasMultiSelection,
-				getSettings,
-			} = select( 'core/block-editor' );
-			const { hasReducedUI } = getSettings();
-			if ( hasReducedUI ) {
-				return true;
-			}
-			const multiSelectedBlockClientIds = getMultiSelectedBlockClientIds();
-			const selectedBlockClientId = getSelectedBlockClientId();
-			return hasMultiSelection()
-				? multiSelectedBlockClientIds.includes( clientId )
-				: clientId === selectedBlockClientId;
-		},
-		[ clientId ]
-	);
-
-	function focusClosestTabbable( event ) {
-		const { clientX, clientY, target } = event;
-
-		// Only handle click on the wrapper specifically, and not an event
-		// bubbled from the inserter itself.
-		if ( target !== ref.current ) {
-			return;
-		}
-
-		const { ownerDocument } = containerRef.current;
-		const targetRect = target.getBoundingClientRect();
-		const isReverse = clientY < targetRect.top + targetRect.height / 2;
-		const blockNode = getBlockDOMNode( clientId, ownerDocument );
-		const container = isReverse ? containerRef.current : blockNode;
-		const closest =
-			getClosestTabbable( blockNode, true, container ) || blockNode;
-		const rect = new window.DOMRect( clientX, clientY, 0, 16 );
-
-		placeCaretAtVerticalEdge( closest, isReverse, rect, false );
```

Comment:
I think we will want to bring this logic back though it probably needs adaptation for horizontal contexts. See #28682